src/gcp/rules.ts

```diff
@@ -97,6 +97,20 @@ export async function listRulesets(
   return _handleErrorResponse(response);
 }
 
+/**
+ * Lists all the rulesets for the given project. May require many network requests.
+ */
+export async function listAllRulesets(projectId: string): Promise<object[]> {
+  let pageToken = undefined;
+  let rulesets: object[] = [];
+  do {
```

Comment:
I literally can't think of another time I've seen a `do { } while ()` in JS code...I don't necessarily have a problem with it but it makes me slightly uneasy :grimacing:

Reply:
I basically only ever use do/while loops when doing pagination.